alibi/explainers/tests/test_pd_variance.py

```diff
@@ -243,25 +251,384 @@ def test_interaction_num_cat(rf_classifier, pd_pdv_explainers, features, adult_d
+        'grid_resolution': 4,
+        'feature_names': ['f_0', 'f_1', 'f_2', 'f_3'],
+        'categorical_names': {
+            2: [0, 1, 2, 3, 4],
+            3: [0, 1, 2, 3, 4, 5, 6]
+        },
+        'target_names': ['c_0', 'c_1'],
+        'method': 'interaction'
+    }
+    data['feature_deciles'] = [
+        [
+            np.array([-2.319, -2.032, -1.744, -1.456, -1.169, -0.881, -0.593, -0.305, -0.018, 0.269, 0.557]),
+            np.array([0.080, 0.217, 0.354, 0.491, 0.628, 0.765, 0.902, 1.039, 1.175, 1.312, 1.449])
+        ],
+        [
+            np.array([0.080, 0.217, 0.354, 0.491, 0.628, 0.765, 0.902, 1.039, 1.175, 1.312, 1.449]),
+            None
+        ],
+        [
+            None,
+            None
+        ]
+    ]
+    data['pd_values'] = [
+        np.array([
+            [[-157.035, -39.254], [46.342, 164.122]],
+            [[-204.567, -158.093], [52.533, 99.006]],
+            [[-82.799, -81.085], [20.010, 21.725]]
+        ]),
+        np.array([
+            [[-55.346, -55.346], [62.434, 62.434]],
+            [[-76.017, -51.231], [-12.543, -29.543]],
+            [[-31.394, -67.631], [-29.679, -4.852]]
+        ]),
+        np.array([
+            [[3.543, 5.143], [6.123, 12.755]],
+            [[-52.214, -1.523], [-27.732, -43.564]],
+            [[-13.537, -35.537], [-5.123, -30.537]]
+        ])
+    ]
+    data['feature_values'] = [
+        [
+            np.array([-2.319, 0.557]),
+            np.array([0.080, 1.449])
+        ],
+        [
+            np.array([0.080, 1.449]),
+            np.array([1., 4.])
+        ],
+        [
+            np.array([1., 4.]),
+            np.array([0., 4.])
+        ]
+    ]
+    data['feature_names'] = [('f_0', 'f_1'), ('f_1', 'f_2'), ('f_2', 'f_3')]
+    data['feature_importance'] = None
+    data['feature_interaction'] = np.array([
+        [1., 3., 2.],
+        [5., 1., 4.],
+        [1., 3., 2.]
+    ])
+    data['conditional_importance'] = [
+        [np.array([1., 5., 4.]), np.array([2., 3., 1.])],
+        [np.array([7., 2., 1.]), np.array([3., 2., 1.])],
+        [np.array([6., 3., 3.]), np.array([4., 1., 5.])]
+    ]
+    data['conditional_importance_values'] = [
+        [
+            np.array([[43.879, 24.657], [91.827, 32.861], [29.467, 1.212]]),
+            np.array([[59.551, 143.809], [181.797, 29.227], [91.411, 72.698]])
+        ],
+        [
+            np.array([[72.121, 32.357], [21.731, 13.441], [53.137, 89.321]]),
+            np.array([[83.283, 70.373], [32.861, 34.186], [29.601, 30.791]])
+        ],
+        [
+            np.array([[19.919, 10.809], [14.462, 50.703], [70.373, 35.073]]),
+            np.array([[65.962, 18.605], [11.311, 14.507], [21.231, 24.657]])
+        ]
+    ]
+    return Explanation(meta=meta, data=data)
+
+
+@pytest.mark.parametrize('n_axes', [4, 5, 6, 7])
+@pytest.mark.parametrize('n_cols', [2, 3, 4])
+@pytest.mark.parametrize('exp', [lazy_fixture('explanation_importance')])
```

Comment:
Not sure why this needs to be a lazy fixture since it doesn't have it any parameters, can't test functions just depend on this fixture directly as usual?

Reply:
Yes ... I think initially I intended to use both types of explanations in the test ... I will update it.